File: clases/09_sesion_de_problemas/conjetura_de_collatz/conjetura_de_collatz.cpp
#include <iostream>
using namespace std;

int main(void) {
    int n = 27;
    int pasos = 0;

    int original = n;

    while (n > 1) {
        int m = 0;
        if (n % 2 == 0) {
            m = n / 2;
            cout << n << " es par, entonces tomo la mitad: " << m << endl;
        } else {
            m = 3 * n + 1;
            cout << n << " es impar, entonces hago 3n + 1: " << m << endl;
        }
        n = m;
        pasos++;
    }

    cout << "Wow, fue un viaje de " << pasos << " paso(s) desde " << original << " hasta 1.";
    cout <<  " Pero al final llegue. Eso quiere decir que " << original << " es maravilloso.";
    cout << " Me pregunto que numeros no seran maravillosos ..." << endl;;
    return 0;
}

File: tareas/2/24190110/Collatz/conjetura_de_collatz.py
with open('entrada.txt','r') as archivo:
  n=int(archivo.read())
pasos= 0
original = n

with open('salida.txt','w') as archivo2:
  while n>1:

    if n%2==0:
      m=n/2
      archivo2.write(f"{int(n)} es par, estonces tomo la mitad: {int(m)}\n")

    else:
     m=3*n+1
     archivo2.write(f"{int(n)} es impar, entonces hago 3n + 1: {int(m)}\n")
    n=m
    pasos=pasos +1

archivo2.write(f"Wow, fue un viaje de {pasos} paso(s) desde {original} hasta 1.")
archivo2.write(f"Pero al final llegué. Eso quiere decir que {original} es maravilloso.")
archivo2.write("Me pregunto qué números no serán maravillosos ...")
